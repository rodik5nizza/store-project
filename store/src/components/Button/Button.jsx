import React from "react";
import styles from './Button.module.scss'
import PropTypes from "prop-types";

const Button = ({ backgroundColor, text, handleClick, modalNumber }) => {
    
        return (
            <button className={styles.btn} style={ {backgroundColor:backgroundColor} } onClick={ (e) => handleClick(e, modalNumber) }>{ text }</button>
        )
}

Button.propTypes = {
    backgroundColor: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    handleClick: PropTypes.func,
    modalNumber: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
        PropTypes.object
    ]),
}

Button.defaultProps = {
    hendleClick: () => {},
    modalNumber: '',
}

export default Button