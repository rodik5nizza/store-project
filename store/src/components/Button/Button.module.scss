.btn{
    margin-top: 10px;
    min-width: 103px;
    margin: 4px;
    padding-block: 16px;
    padding-inline: 28px;
    font-size: 15px;
    border-radius: 7px;
    color: white;
    cursor: pointer;
    border: none;
}