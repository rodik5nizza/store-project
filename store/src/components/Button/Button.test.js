import Button from './Button';
import {render, screen, fireEvent} from '@testing-library/react';

const clickJest = jest.fn()

describe('Button snapshot testing', () => {
    test('should Button match snapshot', () => {
        const {asFragment} = render(<Button text='Button' backgroundColor='#fff'/>)
        expect(asFragment()).toMatchSnapshot()
    });
})

describe('Heandle click at a Button', () => {
    test('should heandle clixk work', () => {
        render(<Button text='Button' backgroundColor='#fff' handleClick={clickJest}/>)
        fireEvent.click(screen.getByText('Button'))
        expect(clickJest).toHaveBeenCalled()
    });
})