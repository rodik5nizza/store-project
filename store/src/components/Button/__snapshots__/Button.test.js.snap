// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot testing should Button match snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn"
    style="background-color: rgb(255, 255, 255);"
  >
    Button
  </button>
</DocumentFragment>
`;
